File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjCSharp/Model/Employee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Model
{
    public class Employee : Entity<long>
    {
        public String Firstname { get; set; }
        public String Lastname { get; set; }
        public String Username { get; set; }
        public String HashedPassword { get; set; }

        public Employee(): base(0)
        {
            Firstname = "";
            Lastname = "";
            Username = "";
            HashedPassword = "";
        }

        public Employee(string username, string pass): base(0)
        {
            Firstname = "";
            Lastname = "";
            Username = username;
            HashedPassword = pass;
        }

        public Employee(string firstname, string lastname, string username, string hashedPassword): base(0)
        {
            Firstname = firstname;
            Lastname = lastname;
            Username = username;
            HashedPassword = hashedPassword;
        }

        public override string ToString()
        {
            return "Employee: " + base.Id + " " + Firstname + " " + Lastname + " " + Username;
        }

        public override bool Equals(object obj)
        {
            return obj is Employee employee &&
                   Id == employee.Id &&
                   Firstname == employee.Firstname &&
                   Lastname == employee.Lastname &&
                   Username == employee.Username &&
                   HashedPassword == employee.HashedPassword;
        }

        public override int GetHashCode()
        {
            int hashCode = -374713858;
            hashCode = hashCode * -1521134295 + Id.GetHashCode();
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Firstname);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Lastname);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Username);
            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(HashedPassword);
            return hashCode;
        }
    }
}

File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjJavaFX/Model/src/main/java/app/model/Employee.java
package app.model;

import java.util.Objects;

public class Employee extends Entity<Long> {
    private String firstname;
    private String lastname;
    private String username;
    private String hashedPassword;

    public Employee(){
        this.firstname = "";
        this.lastname = "";
        this.username = "";
        this.hashedPassword = "";
    }

    public Employee(String username, String pass){
        this.username = username;
        this.hashedPassword = pass;
    }

    public Employee(String firstname, String lastname, String username, String hashedPassword) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.username = username;
        this.hashedPassword = hashedPassword;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getHashedPassword() {
        return hashedPassword;
    }

    public void setHashedPassword(String hashedPassword) {
        this.hashedPassword = hashedPassword;
    }

    @Override
    public String toString() {
        return "Employee: " + super.getId() + " " + firstname + " " + lastname + " " + username;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Employee employee = (Employee) o;
        return Objects.equals(firstname, employee.firstname) && Objects.equals(lastname, employee.lastname) &&
                Objects.equals(username, employee.username) && Objects.equals(hashedPassword, employee.hashedPassword);
    }

    @Override
    public int hashCode() {
        return Objects.hash(firstname, lastname, username, hashedPassword);
    }
}

